alacritty/src/display/content.rs

```diff
@@ -478,31 +481,33 @@ impl DerefMut for RegexMatches {
     }
 }
 
-/// Visible regex match tracking.
+/// Visible hint match with current index.
 #[derive(Default)]
-struct Regex<'a> {
+struct TrackedHintMatches<'a> {
     /// All visible matches.
-    matches: Cow<'a, RegexMatches>,
+    matches: Cow<'a, [HintMatchLocation]>,
 
     /// Index of the last match checked.
     index: usize,
 }
 
-impl<'a> Regex<'a> {
-    /// Create a new renderable regex iterator.
+impl<'a> TrackedHintMatches<'a> {
+    /// Create a new renderable iterator.
     fn new<T>(term: &Term<T>, dfas: &RegexSearch) -> Self {
-        let matches = Cow::Owned(RegexMatches::new(term, dfas));
-        Self { index: 0, matches }
+        let matches = RegexMatches::new(term, dfas);
+        let matches =
+            matches.0.into_iter().map(|bounds| HintMatchLocation::Regex { bounds }).collect();
```

Comment:
This seems rather ugly, mapping everything and recollecting just to convert it to the right wrapper type.